homeassistant/components/motion_blinds/sensor.py

```diff
@@ -0,0 +1,178 @@
+    motion_gateway = hass.data[DOMAIN][config_entry.entry_id][KEY_GATEWAY]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
+
+    for blind in motion_gateway.device_list.values():
+        entities.append(MotionSignalStrengthSensor(coordinator, blind, TYPE_BLIND))
+        if blind.type == BlindType.TopDownBottomUp:
+            entities.append(MotionTDBUBatterySensor(coordinator, blind, "Bottom"))
+            entities.append(MotionTDBUBatterySensor(coordinator, blind, "Top"))
+        elif blind.battery_voltage > 0:
+            # Only add battery powered blinds
+            entities.append(MotionBatterySensor(coordinator, blind))
+
+    entities.append(
+        MotionSignalStrengthSensor(coordinator, motion_gateway, TYPE_GATEWAY)
+    )
+
+    async_add_entities(entities)
+
+
+class MotionBatterySensor(CoordinatorEntity, Entity):
+    """
+    Representation of a Motion Battery Sensor.
+
+    Updates are done by the cover platform.
+    """
+
+    def __init__(self, coordinator, blind):
+        """Initialize the Motion Battery Sensor."""
+        super().__init__(coordinator)
+
+        self._blind = blind
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return f"{self._blind.mac}-battery"
+
+    @property
+    def device_info(self):
+        """Return the device info of the blind."""
+        return {"identifiers": {(DOMAIN, self._blind.mac)}}
+
+    @property
+    def name(self):
+        """Return the name of the blind battery sensor."""
+        return f"{self._blind.blind_type}-battery-{self._blind.mac}"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return PERCENTAGE
+
+    @property
+    def device_class(self):
+        """Return the device class of this entity."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._blind.battery_level
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        attributes = {}
+        attributes[ATTR_BATTERY_VOLTAGE] = self._blind.battery_voltage
+        return attributes
+
+
+class MotionTDBUBatterySensor(MotionBatterySensor):
+    """
+    Representation of a Motion Battery Sensor for a Top Down Bottom Up blind.
+
+    Updates are done by the cover platform.
+    """
+
+    def __init__(self, coordinator, blind, motor):
+        """Initialize the Motion Battery Sensor."""
+        super().__init__(coordinator, blind)
+
+        self._motor = motor
+
+    @property
+    def unique_id(self):
+        """Return the unique id of the blind."""
+        return f"{self._blind.mac}-{self._motor}-battery"
+
+    @property
+    def name(self):
+        """Return the name of the blind battery sensor."""
+        return f"{self._blind.blind_type}-{self._motor}-battery-{self._blind.mac}"
```

Comment:
Is there a more human readable name that can be used?

Reply:
see my comment above